lib/ntlmauth/ntlmauth.cc

```diff
@@ -108,10 +108,19 @@ ntlm_fetch_string(const ntlmhdr *packet, const int32_t packet_size, const strhdr
     int32_t o = le32toh(str->offset);
     // debug("ntlm_fetch_string(plength=%d,l=%d,o=%d)\n",packet_size,l,o);
 
-    if (l < 0 || l > NTLM_MAX_FIELD_LENGTH || o + l > packet_size || o == 0) {
-        debug("ntlm_fetch_string: insane data (pkt-sz: %d, fetch len: %d, offset: %d)\n", packet_size,l,o);
+    if (l < 0 || NTLM_MAX_FIELD_LENGTH < l) {
+        debug("ntlm_fetch_string: insane string length (pkt-sz: %d, fetch len: %d, offset: %d)\n", packet_size,l,o);
```

Comment:
Here and on other new/changed debug() lines, `%d` codes do not match actual argument types (but this bug is present in the official code as well).

Reply:
```%d``` is correct for printing 16/32-bit signed variables. It does not become a problem unless the types are converted to unsigned per the earlier request (which is not the case in current code).

Reply:
> `%d` is correct for printing 16/32-bit signed variables.

I disagree: Lying about small fixed-size argument types "works" in Squid environments due to [default argument promotion](https://en.cppreference.com/w/cpp/language/variadic_arguments), but it is not "correct" because those promotions are not guaranteed to have the right effect for fixed-size types (e.g., `int32_t` is not guaranteed to be promoted to `int` because `int` is not guaranteed to even have 32 bits).

The best solution here is to get rid of printf() style itself, of course, but that requires a lot of work.